Problem 2
If I want to know the ownership, the size, and the permissions of a file, I have to obtain in the first place a list of the files included in a folder, using the command ls -l. To obtain the size in an appropiate format I add a -h. The commands tail and head, and the use of pipes (|) allows me to select and obtain such information from a file of my interest.  
As we can see in the below cell, the owner of the file Locus_Germplasm_Phenotype_20130122.txt is osboxes, as it is shown in column 2. In addition, as we can see in column 4, the size of the file is 1.2 Megabytes. 


In [3]:
ls -lh | tail -2 | head -1

-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  5 07:52 Locus_Germplasm_Phenotype_20130122.txt


The permissions on this file are reflected in column 1. According to the different characters displayed, this file is not a directory; it can be read (r) and written (w) by the owner and the group, and it can be read by anyone 

Problem 3
To get only the first lines of a file it exists the command head followed by a - and a number, to choose the number of lines I want to see. Therefore to display the header line I will type head -1 name_file.txt. 

In [4]:
head -1 Locus_Germplasm_Phenotype_20130122.txt

Locus_name)	Germplasm_name	phenotype	pubmed_id


Problem 4
To find out the total number of lines in a file I have to use the command wc -l.

In [5]:
wc -l Locus_Germplasm_Phenotype_20130122.txt

7216 Locus_Germplasm_Phenotype_20130122.txt


As we can see the Locus Germplasm file has 7216 lines. 

Problem 5
With the command > new_file, I can rewrite in a different file the content of an existing document. In fact, I can use it together with other commands as head or tail to choose the lines of the files we want to rewrite. In this case I want to create a document that contains only the data lines and exlude the header line. To achieve that, I have to type tail -7215 to select all the data lines and then use > Data_Only.csv to rewrite them.      

In [6]:
tail -7215 Locus_Germplasm_Phenotype_20130122.txt > Data_Only.csv
wc -l Data_Only.csv
head -5 Data_Only.csv

7215 Data_Only.csv
AT1G01040	CS3828	Increased abundance of miRNA precursors.	17369351
AT1G01040	CS3828	indeterminate floral meristems, producing extra whorls of stamens and an indefinite number of carpels; unregulated cell division in the center of the flower with normal floral organ identity; plants also show other developmental defects including absence of axillary inflorescence meristems and abnormally shaped cotyledons, leaves, sepals, stamens and carpels	10556049
AT1G01060	lhy-101	The mutant plants are hypersensitive to both FRc and Rc light treatments in hypocotyl elongation and exhibits a small reciprocal enlargement in cotyledon area, albeit not statistically significant.	16891401
AT1G01140	SALK_058629	hypersensitive to low potassium media	17486125
AT1G01220	SALK_012400C	fkgp-1 mutants have about 40 times more L-fucose than wild type Arabidopsis plants, but the levels of other monosaccharides do not appear to differ significantly in the mutants. No obvious phenotypic abnormalit

The commands wc -l and head -5 prove that the code has worked as expected and that the data lines have been rewritten in a new file. 

Problem 6

As I know by the previous problems, the only column with written words is the column with the phenotypes of the plants. Therefore I can use the command grep -no '\root' to know the lines where I can find this specific word. 

In [7]:
grep -no '\root' Locus_Germplasm_Phenotype_20130122.txt

23:root
23:root
23:root
24:root
24:root
24:root
24:root
24:root
26:root
26:root
90:root
90:root
90:root
91:root
91:root
91:root
91:root
91:root
92:root
93:root
94:root
95:root
144:root
144:root
144:root
145:root
145:root
145:root
150:root
150:root
150:root
151:root
151:root
151:root
152:root
152:root
154:root
156:root
180:root
180:root
185:root
185:root
185:root
190:root
190:root
190:root
211:root
224:root
226:root
234:root
234:root
235:root
235:root
251:root
251:root
253:root
255:root
256:root
256:root
257:root
289:root
294:root
336:root
340:root
358:root
358:root
383:root
383:root
383:root
391:root
394:root
410:root
430:root
435:root
436:root
477:root
480:root
496:root
496:root
520:root
521:root
521:root
521:root
535:root
535:root
536:root
536:root
545:root
553:root
553:root
553:root
553:root
553:root
559:root
559:root
563:root
564:root
566:root
566:root
566:root
566:root
567:root
567:root
567:root
567:root
569:root
569:root
571:root
571:root
571:root
571:root
572:root
572:root
572:r

621:root
646:root
646:root
647:root
647:root
647:root
647:root
675:root
675:root
687:root
687:root
701:root
702:root
707:root
715:root
716:root
792:root
792:root
792:root
803:root
803:root
804:root
805:root
805:root
805:root
805:root
810:root
810:root
811:root
812:root
815:root
821:root
822:root
822:root
825:root
847:root
860:root
860:root
861:root
862:root
872:root
879:root
880:root
880:root
882:root
894:root
894:root
896:root
906:root
906:root
906:root
906:root
906:root
906:root
906:root
908:root
913:root
913:root
914:root
963:root
963:root
963:root
964:root
964:root
965:root
966:root
966:root
966:root
966:root
983:root
983:root
1008:root
1010:root
1014:root
1016:root
1027:root
1028:root
1029:root
1043:root
1046:root
1047:root
1079:root
1079:root
1080:root
1081:root
1084:root
1122:root
1124:root
1125:root
1125:root
1126:root
1162:root
1163:root
1163:root
1199:root
1228:root
1229:root
1234:root
1234:root
1237:root
1238:root
1239:root
1239:root
1240:root
1241:root
1249:root
1250:root
1

1365:root
1366:root
1400:root
1401:root
1402:root
1403:root
1407:root
1410:root
1413:root
1413:root
1419:root
1420:root
1424:root
1458:root
1460:root
1487:root
1489:root
1491:root
1501:root
1508:root
1509:root
1509:root
1543:root
1587:root
1636:root
1636:root
1636:root
1643:root
1656:root
1656:root
1656:root
1657:root
1658:root
1658:root
1659:root
1660:root
1660:root
1661:root
1662:root
1668:root
1668:root
1695:root
1695:root
1695:root
1702:root
1705:root
1707:root
1717:root
1718:root
1724:root
1724:root
1724:root
1724:root
1724:root
1727:root
1727:root
1727:root
1740:root
1747:root
1749:root
1749:root
1749:root
1795:root
1833:root
1833:root
1833:root
1833:root
1834:root
1834:root
1836:root
1844:root
1845:root
1855:root
1922:root
1923:root
1926:root
1926:root
1928:root
1928:root
1929:root
1930:root
1930:root
1930:root
1930:root
1935:root
1935:root
1936:root
1937:root
1948:root
1953:root
1956:root
1961:root
1965:root
1965:root
1968:root
1969:root
1969:root
1971:root
1975:root
1975:root


2110:root
2161:root
2161:root
2163:root
2163:root
2186:root
2194:root
2207:root
2220:root
2244:root
2244:root
2244:root
2248:root
2248:root
2249:root
2249:root
2250:root
2250:root
2251:root
2251:root
2251:root
2266:root
2279:root
2296:root
2355:root
2360:root
2361:root
2371:root
2372:root
2373:root
2373:root
2374:root
2377:root
2385:root
2386:root
2387:root
2387:root
2389:root
2391:root
2391:root
2394:root
2394:root
2395:root
2395:root
2419:root
2421:root
2468:root
2468:root
2481:root
2482:root
2483:root
2484:root
2485:root
2486:root
2493:root
2526:root
2534:root
2537:root
2541:root
2541:root
2551:root
2600:root
2600:root
2605:root
2605:root
2605:root
2618:root
2667:root
2668:root
2670:root
2695:root
2701:root
2727:root
2727:root
2731:root
2732:root
2777:root
2777:root
2777:root
2788:root
2789:root
2791:root
2798:root
2798:root
2798:root
2800:root
2818:root
2818:root
2819:root
2838:root
2852:root
2855:root
2857:root
2869:root
2871:root
2876:root
2887:root
2887:root
2890:root
2908:root


3133:root
3133:root
3133:root
3133:root
3133:root
3134:root
3134:root
3134:root
3134:root
3134:root
3138:root
3143:root
3143:root
3174:root
3179:root
3223:root
3261:root
3270:root
3273:root
3279:root
3286:root
3303:root
3304:root
3304:root
3307:root
3307:root
3308:root
3308:root
3308:root
3316:root
3317:root
3318:root
3319:root
3320:root
3351:root
3355:root
3355:root
3355:root
3355:root
3356:root
3356:root
3356:root
3356:root
3359:root
3375:root
3377:root
3378:root
3379:root
3380:root
3380:root
3386:root
3386:root
3386:root
3386:root
3387:root
3387:root
3387:root
3387:root
3390:root
3391:root
3450:root
3453:root
3453:root
3453:root
3455:root
3458:root
3458:root
3458:root
3462:root
3462:root
3462:root
3466:root
3466:root
3466:root
3470:root
3470:root
3470:root
3473:root
3476:root
3480:root
3482:root
3483:root
3502:root
3502:root
3507:root
3525:root
3526:root
3526:root
3527:root
3542:root
3542:root
3543:root
3544:root
3545:root
3548:root
3548:root
3549:root
3550:root
3550:root
3551:root


3868:root
3868:root
3880:root
3880:root
3880:root
3881:root
3881:root
3885:root
3885:root
3885:root
3886:root
3886:root
3900:root
3904:root
3904:root
3904:root
3907:root
3907:root
3910:root
3910:root
3910:root
3918:root
3936:root
3936:root
3943:root
4032:root
4032:root
4033:root
4055:root
4057:root
4058:root
4060:root
4072:root
4073:root
4074:root
4109:root
4110:root
4146:root
4148:root
4149:root
4150:root
4189:root
4189:root
4189:root
4190:root
4190:root
4190:root
4191:root
4191:root
4191:root
4274:root
4275:root
4281:root
4281:root
4282:root
4282:root
4305:root
4313:root
4328:root
4328:root
4339:root
4345:root
4353:root
4354:root
4364:root
4371:root
4372:root
4372:root
4388:root
4388:root
4391:root
4392:root
4392:root
4439:root
4440:root
4440:root
4441:root
4441:root
4441:root
4450:root
4453:root
4455:root
4471:root
4514:root
4514:root
4515:root
4515:root
4515:root
4515:root
4515:root
4515:root
4516:root
4516:root
4517:root
4517:root
4518:root
4518:root
4519:root
4519:root
4520:root


4985:root
4985:root
4990:root
4990:root
4990:root
4990:root
5007:root
5007:root
5008:root
5025:root
5026:root
5028:root
5028:root
5028:root
5029:root
5030:root
5082:root
5085:root
5089:root
5090:root
5091:root
5091:root
5091:root
5091:root
5092:root
5092:root
5092:root
5092:root
5092:root
5092:root
5093:root
5102:root
5118:root
5135:root
5164:root
5182:root
5190:root
5264:root
5265:root
5265:root
5275:root
5276:root
5325:root
5326:root
5327:root
5344:root
5347:root
5348:root
5349:root
5349:root
5352:root
5353:root
5419:root
5419:root
5419:root
5419:root
5421:root
5421:root
5421:root
5438:root
5439:root
5439:root
5448:root
5451:root
5452:root
5452:root
5453:root
5453:root
5456:root
5460:root
5462:root
5463:root
5465:root
5466:root
5473:root
5475:root
5584:root
5584:root
5584:root
5586:root
5586:root
5587:root
5588:root
5589:root
5591:root
5591:root
5592:root
5592:root
5602:root
5616:root
5617:root
5617:root
5618:root
5667:root
5667:root
5668:root
5668:root
5672:root
5672:root
5673:root


5924:root
5954:root
5968:root
5968:root
5972:root
5972:root
5972:root
5973:root
5973:root
5973:root
5974:root
5974:root
5974:root
5975:root
5975:root
5976:root
5976:root
5977:root
5977:root
5978:root
5978:root
5978:root
5980:root
5980:root
5980:root
5980:root
5981:root
5981:root
5981:root
5981:root
5983:root
5983:root
5985:root
5985:root
5985:root
5985:root
5986:root
5986:root
5986:root
5986:root
5988:root
5988:root
5990:root
5990:root
5990:root
5990:root
5991:root
5991:root
5991:root
5991:root
5994:root
5994:root
5994:root
5994:root
5995:root
5995:root
5995:root
5995:root
5998:root
5998:root
5998:root
5998:root
5999:root
5999:root
5999:root
5999:root
6002:root
6002:root
6002:root
6002:root
6003:root
6003:root
6003:root
6003:root
6005:root
6005:root
6006:root
6006:root
6012:root
6018:root
6026:root
6035:root
6046:root
6047:root
6047:root
6050:root
6050:root
6050:root
6157:root
6164:root
6165:root
6167:root
6167:root
6167:root
6167:root
6169:root
6178:root
6180:root
6186:root
6187:root


6389:root
6394:root
6440:root
6440:root
6466:root
6466:root
6467:root
6467:root
6468:root
6468:root
6469:root
6470:root
6476:root
6477:root
6478:root
6479:root
6487:root
6488:root
6489:root
6500:root
6503:root
6521:root
6524:root
6552:root
6558:root
6566:root
6571:root
6572:root
6583:root
6583:root
6583:root
6584:root
6584:root
6584:root
6585:root
6586:root
6586:root
6586:root
6592:root
6596:root
6597:root
6602:root
6618:root
6620:root
6621:root
6624:root
6627:root
6628:root
6629:root
6629:root
6629:root
6630:root
6632:root
6633:root
6634:root
6636:root
6636:root
6649:root
6759:root
6775:root
6784:root
6786:root
6787:root
6792:root
6792:root
6794:root
6796:root
6800:root
6800:root
6802:root
6807:root
6807:root
6809:root
6814:root
6814:root
6816:root
6821:root
6821:root
6824:root
6835:root
6838:root
6838:root
6838:root
6838:root
6838:root
6842:root
6842:root
6855:root
6856:root
6859:root
6868:root
6869:root
6881:root
6881:root
6904:root
6904:root
6904:root
6904:root
6904:root
6904:root


7191:root
7192:root
7192:root
7193:root
7193:root
7194:root
7194:root
7195:root
7202:root
7202:root
7202:root
7203:root


Problem 7
First of all, I used the same command (except for the -o) to select the lines that have a phenotype associated with the word root in the file. Then I used a pipe and the comand grep -E -in '\AT[0-9]G[0-9]{5}' to obtain the Locus codes included in those lines.

In [8]:
grep -n '\root' Locus_Germplasm_Phenotype_20130122.txt | grep -E -in '\AT[0-9]G[0-9]{5}'

1:23:AT1G01550	bps1-2	Growth of bps1-2 mutants on CPTA-supplemented medium resulted in partial rescue of both leaf and root defects. bps1-2 mutants grown on control growth medium have small radialized leaves with very little vascular tissue, and very short misshapen knotted-looking roots. By contrast, bps1-2 mutants grown on CPTA-supplemented medium produced larger flattened leaves that contained primary and secondary veins, and smooth elongated roots.	17217459
2:24:AT1G01550	bps1-2	Mutant roots were abnormal; primary and lateral roots were short, root hairs formed close to the root apex, and root defects were most severe when plants were grown at low temperature.	15458645
3:26:AT1G01550	bps1-2	Under high light conditions (approximately 200 &mu;E.m-2.sec-1), CPTA-treated seedlings were completely photobleached and mutants showed rescue of both leaf and root development. Under moderate (approximately 100 &mu;E.m-2.sec-1) and low (30 &mu;E.m-2.sec-1) light intensities, seedlings are inco

26:253:AT1G08260	tilted1-4	Twenty-five percent of embryos show abnormal cell divisions during embryogenesis-large cells- decreased rate of cell division. Division of the hypohyseal cell abnormal leading to abnormal placement of the root pole. Homozygotes show delayed flowering, abnormal floraly phyllotaxy and abnormal ovules.	16278345
27:255:AT1G08370	dcp1-2	The homozygous progeny is seedling lethal, showed arrested postembryonic development including cotyledon expansion, development of vascular networks, root elongation, and shoot development.	17485080
28:256:AT1G08430	Atalm1-KO	On Al3+ medium, the mutant has significantly shorter roots than the Col wild type. In the absence of aluminum stress, the roots of mutant and wild type grew similarly.	16740662
29:257:AT1G08430	Atalm1-KO	The mutant exhibits very low levels of root  growth and malate release in the presence of  Al (0.2 nmol plant-1 24 h-1).
30:289:AT1G09100	rpt5b	Insensitive to high-glucose repression of cotyledon expansion, ch

51:559:AT1G19220	CS24617	The primary roots of the mutant produce as many lateral roots as the wild type.	15659631
52:563:AT1G19220	CS24617	no obvious auxin-related growth phenotype, but roots show mild auxin resistance	15659631
53:564:AT1G19220	CS24618	no obvious auxin-related growth phenotype, but roots show mild auxin resistance	15659631
54:567:AT1G19220	CS24625	The phenotype of the double mutant is most obvious at its seedling stage, with its most prominent phenotype being severely impaired  lateral root formation. Its primary roots fail to produce lateral roots in 2-week-old seedlings. However, double mutant seedlings start to generate several lateral roots after ~2 weeks of growth, and their morphological appearance is normal.	15659631
55:569:AT1G19220	CS24625	nph4-1 arf19 double mutant;  agravitropic response in both hypocotyls and roots; impaired phototropic response in hypocotyls; impaired lateral root formation; small plant size; small and epinastic rosette leaves; reduced aux

78:810:AT1G27320	ahk2-1cu ahk3-1cu ahk4-1cu	Normal vascular system in adventitious roots, although the hypocotyl and primary root develop aborted vascular tissues.	16357038
79:811:AT1G27320	ahk2-1cu ahk3-1cu ahk4-1cu	Postembryonic growth of roots (primary and lateral) severely impaired.	11479382
80:812:AT1G27320	ahk2-1cu ahk3-1cu ahk4-1cu	Reduced seedling size with small cotyledons, slightly shortened hypocotyls (65% of wildtype) and significantly reduced root length.	11479382
81:815:AT1G27320	ahk2-1ms ahk3-1ms	Normal roots.	
82:821:AT1G27320	ahk2-2tk ahk3-3	Normal roots.	
83:822:AT1G27320	ahk2-2tk ahk3-3	Reduced sensitivity to cytokinin in root growth assay (exogenous application of cytokinins inhibits wildtype root elongation).	
84:825:AT1G27320	ahk2-5 ahk3-7	Displays secondary lateral root branching (not seen in wild-type).	16361392
85:847:AT1G27320	ahk2-5 ahk3-7 cre1-2	Increased seed size, largely due to increased size of the embryo. Embryonic root epidermal cells increased 15% in 

105:1010:AT1G33280	smb-3; brn1-1; brn2-1	Root cap cells appear as a mass lacking uniform morphology. Cells can be highly elongated and twisted.   Cells fail to stop elongation and to  adopt the rigid, rectangular morphology of COL cells.   When compared with smb-3, the triple mutant has thesame number of complete LRC layers, and the appearance of the cells in those layers closely resembles that in smb-3. However, partial remains of older LRC layers remain attached to the meristem in the triple mutant, giving it a messy appearance. Remains of old LRC cap layers can be  found still attached to the root along its whole length.
106:1014:AT1G33290	brn1-1; brn2-1	Columella root cap cells fail to detach.	20197506
107:1016:AT1G33290	smb-3; brn1-1; brn2-1	Root cap cells appear as a mass lacking uniform morphology. Cells can be highly elongated and twisted.   Cells fail to stop elongation and to  adopt the rigid, rectangular morphology of COL cells.   When compared with smb-3, the triple mutant 

135:1253:AT1G54990	CS8040	auxin resistant; narrow, irregular rosette leaves, slightly curled around leaf axis; roots elongate on auxin-containing medium; defective root gravitropism; reduced number of lateral roots - greater reduction than for either single mutant; dwarf, bushy plants; reduced plant height; ethylene resistant; reduced fertility.	7704045
136:1254:AT1G54990	axr4-1 aux1-7	auxin resistant; narrow, irregular rosette leaves, slightly curled around leaf axis; roots elongate on auxin-containing medium; defective root gravitropism; reduced number of lateral roots - greater reduction than for either single mutant; dwarf, bushy plants; reduced plant height; ethylene resistant; reduced fertility.	7704045
137:1257:AT1G55020	SALK_059431	developed more emergent (stage VIII) and lateral roots at 10 d. has a moderate increase in the length of the primary root.	17369372
138:1262:AT1G55250	hub2-1	Reduction in the rosette biomass to 43 and 42% of fresh and dry weight of Col, respectively 

162:1402:AT1G64440	reb1-1	Decreased root length (67% of that of wildtype) when grown on D-galactose-free medium.	12419184
163:1403:AT1G64440	reb1-1	Less galactose in root cell wall compared to wildtype. No difference in shoot cell wall monosaccharide composition.	16500990
164:1407:AT1G64440	reb1-1	Structural changes of xyloglucan in roots, not in shoots.	16500990
165:1410:AT1G64440	reb1-2	At the restrictive temperature, epidermal bulging is observed and roots fail to grow straight in the elongation zone although this mutation does not appear to abolish gravitropic sensitivity.	
166:1413:AT1G64520	rpn12a-1 (C24)	The rpn12a-1 mutant produces both the WT RPN12a protein and a RPN12a(1-186):ARP(exon2):NPTII fusion protein. The mutant form of the protein can coelute with subunits of the 26S proteasome from the regulatory and core particles and there is no increase in total steady state levels of ubiquitinated proteins in homozygous mutant seedlings. This mutant has seedling growth defects, e

184:1658:AT1G70560	wei8-1 tar2-1	The roots of this mutant show almost no sensitivity to the ethylene precursor ACC, but sensitivity is largely restored by the addition of low levels of IAA (but not tryptophan). ACC-triggered dark-grown apical hook formation is also compromised in this mutant, and it is not restored by the addition of IAA. Root gravitropism is more disrupted in this double mutant than in either single mutant. The mutant plants are shorter than wild type, have reduced venation in their leaves, exhibit reduced apical dominance, and produce abnormal flowers. DR5:GUS expression is also diminished in these double mutants, consistent with a ~50% reduction in IAA levels in these mutants. Root meristematic cells differentiate in this mutant, leading to a loss of the stem cell niche and the cessation of root growth.	18394997
185:1659:AT1G70560	wei8-1 tar2-2	ACC-triggered dark-grown apical hook formation is compromised in this mutant. Root meristematic cells differentiate in this

208:1926:AT2G01830	CS6563	Reduced sensitivity to cytokinin in root growth assay (exogenous application of cytokinins inhibits wildtype root elongation).	
209:1928:AT2G01830	ahk2-1cu ahk3-1cu ahk4-1cu	Aborted vascular system containing few protoxylem cells in the primary root. Normal adventitious root-vascular system.	16357038
210:1929:AT2G01830	ahk2-1cu ahk3-1cu ahk4-1cu	Adventitious roots are normal.	16357038
211:1930:AT2G01830	ahk2-1cu ahk3-1cu ahk4-1cu	Does not produce lateral roots from the primary root. However, 10 days after germination, lateral roots are formed at the upper part of the hypocotyl rather than at the root-hypocotyl junction.	16357038
212:1935:AT2G01830	ahk2-1cu ahk3-1cu ahk4-1cu	Normal vascular system in adventitious roots, although the hypocotyl and primary root develop aborted vascular tissues.	16357038
213:1936:AT2G01830	ahk2-1cu ahk3-1cu ahk4-1cu	Postembryonic growth of roots (primary and lateral) severely impaired.	11479382
214:1937:AT2G01830	ahk2-1cu ahk3-1cu

253:2099:AT2G02950	pks1-1	No obvious root growth phenotypes.	14615593
254:2105:AT2G03680	CS6546	Defective in directional cell elongation processes; abnormal cortical microtubule function; exhibits right-handed helical growth in roots and etiolated hypocotyls; epidermal cell files of roots are twisted to form right-handed helices; on vertically oriented hard agar plates, roots grow to the right when viewed from above the agar plates; this skewed root growth is driven by the friction between agar surface and helical epidermal cell files; phenotype is enhanced under the conditions that accelerate cell elongation, under such conditions, epidermal cells undergo isotropic cell expansion, resulting in spherically shaped cells protruding from the organ surface.	15084720
255:2106:AT2G03680	CS6547	Defective in directional cell elongation processes; abnormal cortical microtubule function; exhibits right-handed helical growth in roots and etiolated hypocotyls; epidermal cell files of roots are twi

273:2361:AT2G24790	col3	Mutant plants have longer hypocotyls in red  light and in short days. Unlike constans, the col3 mutant flowers early and shows a reduced number of lateral branches in  short days. The mutant also exhibits reduced formation of lateral roots. The col3 mutation partially suppresses the cop1 and  deetiolated1 (det1) mutations in the dark	16339850
274:2371:AT2G25180	CS6978	Mutant root meristems are larger than those of wild type already at 2 dpg, but they stopped growing after reaching a fixed number of cells at 5 dpg.	17363254
275:2372:AT2G25180	ahk3-3 arr12-2	The root-meristem size of the double mutant was indistinguishable from that of the ahk3 mutant.	17363254
276:2373:AT2G25180	arr1-3 arr10-5 arr12-1	severe primary root abnormalities; premature termination of primary root growth; substantially reduced sensitivity to cytokinin in the hypocotyl growth response assay; smaller rosette diameter; altered chlorophyll and anthocyanin production.	18723577
277:2374:AT2G25

303:2618:AT2G32700	luh-1	Reduced germination (80% of normal), slow initial growth, short roots.	18390806
304:2667:AT2G33880	stip-1	When grown in the absence of sugars plants exhibit growth arrest after germination. Seedling lethal. Shoots exhibit reduced sensitivity to cytokinins but roots appear normal.	20110319
305:2668:AT2G33880	stip-2	Hypomorphic allele identified as an intragenic supressor of stip-d activation tagged allele. Homozyous plants show about 80% seedling lethality. Other phenotypes include hyponastic cotyledon and reduced root size. Expression of WUS is reduced and the domain of STM (a marker of meristems)reduced.Heterozygotes have 25% abnormal embryos with smaller meristems. Phenotype can be rescued with the addition of sucrose in the growth media.	15753038
306:2670:AT2G33880	stip-d ahk2-2 ahk3-3 cre1-12	Small SAM and limited root growth.	20110319
307:2695:AT2G35600	SALK_038885	No visible root phenotype.	16514016
308:2701:AT2G36120	dot1-1	dot1-1 mutants display an open

329:2908:AT2G40890	cyp98A3	background Wassilewskija hypocotyls with shortened length and increased diameter, roots had a swollen aspect with increased initiation from the crown and showed reduced growth and gravitropism   in soil plant maintained the dwarfed phenotype with a rosette never exceeding 1 to 1.5 cm in diameter, developed a bushy miniature rosette of round leaves, growth of cyp98A3 plants was arrested latest at the onset of primary stem development (2 weeks later than wild-type), showed darker leaf coloration , severe alteration of lignin content and composition - mainly H units (95% to 3.5% in wild-type), ectopic lignification in roots
330:2912:AT2G40950	zip17	zip17 mutants exhibit a greater inhibition of primary root elongation in response to NaCl than wild type seedlings. The salt-sensitive phenotype co-segregates with the zip17 T-DNA and can be rescued by a 35S:AtbZIP17 construct. Several salt-responsive genes, such as ATHB-7, that showed a reduced transcriptional respon

352:3132:AT2G47000	pgp4-1	pgp4-1 mutants have 25% longer root hairs than wild type plants.	18156217
353:3133:AT2G47000	pgp4-3	The root lengths of 10-d pgp4 seedlings were 30% shorter than those of the wild type, and the number of lateral roots was also significantly reduced under moderate light levels (100 to 120 mmolm2s1) using sucrose concentrations of 0.5 to 1%. Under high light or on sucrose concentrations >1.5%, the root lengths and number of lateral roots observed were greater than or equal to those of the wild type. No obvious morphological differences between pgp4 and wild-type plants were observed in the aerial parts of seedlings and adult plants. Seedlings exhibited reductions in the rate of root gravitropic bending.	16243904
354:3134:AT2G47000	pgp4-4	The root lengths of 10-d pgp4 seedlings were 30% shorter than those of the wild type, and the number of lateral roots was also significantly reduced under moderate light levels (100 to 120 mmolm2s1) using sucrose concentrations 

377:3359:AT3G08710	trx h9	Roots and leaves of  the homozygous mutants were significantly shorter and smaller  than wild-type. Root tips of 7-day-old mutant seedlings  grown on MS medium with 1% sucrose had a shortened apical  meristem and, cells were more compact, in keeping with noticeably  shorter roots. When grown in soil, mutant  plants were dwarf with small yellowish leaves.
378:3375:AT3G09780	acr4-2/ccr1-3	Increased number of lateral root meristems, greater than acr4 single mutant alone.	18948541
379:3377:AT3G09840	cdc48-1	Seedling lethal.Mutants have disorganized roots.	18660433
380:3378:AT3G09920	pip5k9-d	Concentration of sucrose in the mutant roots are lower than in wild type. The concentration of <i>myo</i>-inositol is increased, as are those of glucose and fructose.	17220200
381:3379:AT3G09920	pip5k9-d	Shortened primary root (~30% compared to wild type). Shortening is more pronounced 6d after germination. Phenotype results from a reduction in cell elongation.	17220200
382:33

405:3596:AT3G18780	act2-1	Mutants have abnormal root hairs that are stunted, bulging or may be branched. The orientation of microtubules is abnormal.	19304937
406:3597:AT3G18780	act2-1/act7-4	Plants are dwarf and reach only 15% of the height of wild type. Roots lack root hairs- the initiate but do not develop further than a bulge. Leaves are small with reduced surface area and fewer lobes. Leaf trichomes are abnormal and contain fewer branches. Also defects in flower morphology, reduced fertility, abnormal inflorescence structure and abnormal silique development.	19304937
407:3598:AT3G18780	act2-1/act8-2	Lacks root hairs.Root hairs initiate but do not develop further.	19304937
408:3606:AT3G19050	pok1-1/pok2-1	At seedling stage pok1/pok2 double mutants differ from wild-type by having smaller cotyledons as well as shorter, wider roots and hypocotyls. Adult plants exhibit a dwarfed stature, but all organs are present, although miniaturized. Plants are fertile but produce reduced numbers o

433:3806:AT3G27580	d6pkl3-2/d6pkl2-2/d6plk1-1/d6pk-1	Quadruple mutant displays defects in many aspects of development. Defects include:abnormal leaves, leaves are narrow and twisted; reduced number of axillary shoots and reduced fertility. In addition, mutants also display single or fused cotyldons, reduced number of lateral roots and are mildy agravitropic.	19168677
434:3846:AT3G27920	aba2-2	Mutant showed a similar sensitivity to 25 mM trehalose as wild type (inhibition of root growth).	17031512
435:3865:AT3G27920	sos3-1 siz1-1	Decrease of macronutrients concentration in the MS salt formulation to 1/20X results in a substantial reduction in primary root growth.	15894620
436:3866:AT3G27920	sos3-1 siz1-1	Mutant grown in greenhouse or growth chamber exhibits reduced shoot and root biomass relative to the wild-type.	15894620
437:3868:AT3G27920	sos3-1 siz1-1	Substantially more pronounced prototypical Pi starvation root architecture responses than sos3-1 or wild-type seedlings. On Pi-limite

458:4109:AT3G53020	CS11180	Defects in apical-basal gynoecium patterning (similar to ett and mp mutants). Mutant plant gynoecia had shorter ovaries and longer gynophores than the wild type, but the total length of the gynoecium is normal before pollination. After pollination, the gynoecium of failed to elongate, resulting in shorter siliques which contained arrested seeds, resulting in lower fertility. The stv1-1 ovules had shorter integuments than wild-type ones, and in some extreme cases, the gametophyte was protruding from the integuments. Mutant plants had defects in vascular system and embryo organization, which are also associated with auxin signaling. Most cotyledons in mutant plants had abnormal vascular patterns that were asymmetric and/or disconnected. Cotyledons were occasionally fused or single, indicating that stv1-1 is associated with defects in embryo patterning. Growth was retarded in both aerial and underground parts of stv1-1 mutants, resulting in plants that were smal

468:4275:AT3G59420	acr4-2/ccr1-3	Increased number of lateral root meristems, greater than acr4 single mutant alone.	18948541
469:4281:AT3G59770	sac9-1	slow growing, small leaves with short petioles, accumulates anthocyanin in the leaves, small and constitutively closed guard cells, shorter primary roots, fewer lateral roots, hypersensitive to osmotic stress, cold stress and high-light stress	15923324
470:4282:AT3G59770	sac9-2	slow growing, small leaves with short petioles, accumulates anthocyanin in the leaves, small and constitutively closed guard cells, shorter primary roots, fewer lateral roots, hypersensitive to osmotic stress, cold stress and high-light stress	15923324
471:4305:AT3G61650	TUBG1-1/tubg1-1 TUBG2-2/tubg2-2	During early seedling growth, cotyledons were small and distorted, and only two leaf primordia were inittaited at the shoot apex. Dome-shape soot apical meristem was distorted. Root growth was inhibited, and root apex was swollen.	16698945
472:4313:AT3G61890	athb12-

499:4562:AT4G10350	brn1-1; brn2-1	Columella root cap cells fail to detach.	20197506
500:4563:AT4G10350	smb-3; brn1-1; brn2-1	Root cap cells appear as a mass lacking uniform morphology. Cells can be highly elongated and twisted.   Cells fail to stop elongation and to  adopt the rigid, rectangular morphology of COL cells.   When compared with smb-3, the triple mutant has thesame number of complete LRC layers, and the appearance of the cells in those layers closely resembles that in smb-3. However, partial remains of older LRC layers remain attached to the meristem in the triple mutant, giving it a messy appearance. Remains of old LRC cap layers can be  found still attached to the root along its whole length.
501:4567:AT4G10380	SALK_122287C	Inhibition root, shoot, flowers and siliques growth when Boron was supplied at concentrations of <10 uM.	16679457
502:4618:AT4G14550	slr-1	Absence of lateral roots. Defective response to auxin stimulus in the root.	11862947
503:4639:AT4G14850	loi1-1	lo

521:4944:AT4G26690	SALK_024208	Homozygotes: Tip growth in almost all root hair cells was blocked due to rupture of the tip.	18718934
522:4946:AT4G26690	shv3-1	Homozygotes: Tip growth in almost all root hair cells was blocked due to rupture of the tip.	18718934
523:4947:AT4G26690	shv3-2;svl1-1	Increased anthocyanin accumulation. Collapsed trichomes.Brown discoloration around endodermis in the hypocotyl and root due to ectopic lignification. Guard cells are larger than WT due to swollen ventral sides. Swollen hypocotyl epidermis with uneven surface. Four-day old dark-grown seedlings have shorter, thicker hypocotyls than WT.   Cell wall analysis: reduced crystalline cellulose. Increased starch content. Neutral sugar content is same as WT. Increases protein. Increased uronic acid and decreased amount of methyl esterification.
524:4954:AT4G27630	gtg1-1 gtg2-1	These double mutants are hyposensitive to ABA in seed germination, post-germination development, root elongation, ABA-induced gene ex

542:5085:AT4G33430	serk3	Mutant exhibited reduced leaf growth (~20%), shorter hypocotyls in the dark, and partial BL insensitivity in root-growth assays.	
543:5089:AT4G33470	hda14	Increased root hair density.	16176989
544:5090:AT4G33480	hda14	Increased root hair density.	16176989
545:5093:AT4G33495	rpd1-2	temperature sensitive, impaired in adventitious root formation in hypocotyl. Similar to rpd1-1 mutation	16407439
546:5102:AT4G33650	drp3a-1	Elongated mitochondria which sometimes form interconnected networks. Enlarged leaf peroxisomes. Elongated root peroxisomes.	18559960
547:5118:AT4G34350	ispH-1	Plants with albino phenotype which exhibit purple-tinted phenotype superimposed on the albino phenotype when grown sucrose-containing medium. ispH-1 plants develop normal roots, rosette leaves and inflorescences.	15863698
548:5135:AT4G35090	cat2-1	Reduced stature, pale green color and reduced number of secondary roots. Decreased catalase activity. Increased H2O2 and increased sensitivity to 

568:5439:AT5G03280	ein2-5	Root growth inhibition by ABA on ein2-5 seedlings increased when the concentrations of ABA were elevated. At the ABA concentrations (0.1-0.25 &#956;M) which stimulate root growth of wild-type plants, root inhibition of the ein2-5  plants started to be observed, and the most profound inhibitory effect was observed at 0.25 &#956;M and higher concentrations of ABA.	17533512
569:5448:AT5G03280	ein2-5	When the mutant was grown on increasing concentrations of ABA, cotyledon greening of ein2-5 seedlings was inhibited dramatically, and the inhibitory effects were increased in a dose-dependent manner. Reduction in percentage of cotyledons turning green of ein2-5 seedlings was more than 10% at as low as 0.1 &mu;M ABA. At 0.5 &mu;M ABA, only about 20% of ein2-5 germinated seedlings turned green while 100% of wild-type seedlings were green at 10 days in light. The root elongation of ein2-5 seedlings was also supersensitive to ABA.
570:5451:AT5G03280	seo2	Compared with the

587:5592:AT5G08550	ilp1-2	Shorter hypocotyls and roots compared to their wild type siblings in darkness.  In the light, plants have shorter hypocotyls and smaller cotyledons compared with the wild type, and there was inhibition of root elongation.  There is also a reduction in the number of 32C cells in 3 day old seedling hypocotyl cells.	17012601
588:5602:AT5G09680	rlf-1	reduced lateral root formation	20230485
589:5616:AT5G09810	act2-1/act7-4	Plants are dwarf and reach only 15% of the height of wild type. Roots lack root hairs- the initiate but do not develop further than a bulge. Leaves are small with reduced surface area and fewer lobes. Leaf trichomes are abnormal and contain fewer branches. Also defects in flower morphology, reduced fertility, abnormal inflorescence structure and abnormal silique development.	19304937
590:5617:AT5G09810	act7-4	Abnormal roots. Roots are shorted and have fewer root hairs (ca. 20% less than wild type). Root epidermal cells have abnormal morphology an

613:5882:AT5G18170	CS3975	deficient in glutamate dehydrogenase; conditional phenotype: impaired root growth and mild shoot chlorosis under intermediate/high inorganic nitrogen conditions, this is exagerated when plants are grown in high concentrations of inorganic nitrogen under suboptimal conditions( e.g. without vitamins); seedling growth is specifically retarded on media containing exogenously supplied inorganic nitrogen.	8643469
614:5883:AT5G18560	puchi-1	shorter visible lateral roots at 9 days after germination. The proximal region of each lateral root in the mutant is significantly swollen and often bent. Growth and morphology of the primary root of the puchi mutant is normal.	17630277
615:5894:AT5G19040	atipt1 atipt 3 atipt 5 atipt7	External application of <i>trans</i>-zeatin partially rescues the growth of aerial parts of the mutant, and reduced its lateral root elongation.	17062755
616:5901:AT5G19040	atipt1 atipt 3 atipt 5 atipt7	Reduced cambial activity and reduced secondary 

635:5995:AT5G20730	CS24628	The phenotype of the double mutant is most obvious at its seedling stage, with its most prominent phenotype being severely impaired  lateral root formation. Its primary roots fail to produce lateral roots in 2-week-old seedlings. However, double mutant seedlings start to generate several lateral roots after ~2 weeks of growth, and their morphological appearance is normal.	15659631
636:5999:AT5G20730	CS24629	The phenotype of the double mutant is most obvious at its seedling stage, with its most prominent phenotype being severely impaired  lateral root formation. Its primary roots fail to produce lateral roots in 2-week-old seedlings. However, double mutant seedlings start to generate several lateral roots after ~2 weeks of growth, and their morphological appearance is normal.	15659631
637:6003:AT5G20730	CS24630	The phenotype of the double mutant is most obvious at its seedling stage, with its most prominent phenotype being severely impaired  lateral root forma

660:6227:AT5G27320	gid1a-1 gid1b-1 gid1c-1	Dramatic reduction in rosette radius and root length compared to wild type (87 and 74% reduction, respectively).	17194763
661:6234:AT5G27320	gid1a-1 gid1c-1	Only slight differences between mutant and wild type plants with regard to rosette radius and root length.	17194763
662:6241:AT5G27320	gid1b-1 gid1c-1	Only slight differences between mutant and wild type plants with regard to rosette radius and root length.	17194763
663:6243:AT5G27320	gid1c-1	Only slight differences between mutant and wild type plants with regard to rosette radius and root length.	17194763
664:6251:AT5G28646	wvd2-1	Inhibition of cell elongation, right-handed root twisting, left-handed leaf petiole twisting, suppression of root-waving, reduced trichome branching, perturbed microtubule organization such as increased longitudinal and oblique positions in root caps.	17319849
665:6266:AT5G35410	CS3863	Do not display altered sensitivity for inhibition of root growth by osmotic s

697:6469:AT5G43940	FLAG_298F11	Unable to grow following germination on nutrient medium in the dark. Unable to grow in the light on growth medium. The mutant seeds germinated, but growth was arrested right after some root elongation and emergence of small cotyledons. The small cotyledons failed to green and the seedlings eventually died. The mutants could however be recovered on soil. Under long day conditions (16 hour light), the mutants had a reduced number of rosette leaves. The leaves were pale green and distorted. Bolts were shorter. After bolting, the plants were highly branched and semi-dwarf under all light conditions. Few seeds setting.	18326829
698:6470:AT5G43940	GABI_315D11	Unable to grow following germination on nutrient medium in the dark. Unable to grow in the light on growth medium. The mutant seeds germinated, but growth was arrested right after some root elongation and emergence of small cotyledons. The small cotyledons failed to green and the seedlings eventually died.

722:6620:AT5G49190	sus2 (SALK_076303)	The amounts of glucose, fructose, sucrose, cellulose (roots) and starch in the mutant were not statistically significantly different  from those of the equivalent wild-type lines grown under the same conditions at the same time.	17257168
723:6621:AT5G49190	sus2 sus3	Activity of SUS was reduced by about one-third in roots of the double mutant relative to Col-0, but otherwise the mutant was indistinguishable from Col-0.	17257168
724:6624:AT5G49270	der9	Root hairs have mostly very short root hairs, indicating a defect in the transition to rapid tip growth, which is required to fully develop the hair structure.	15863435
725:6627:AT5G49680	kip-1	Pollen tube growth was altered. Pollen tubes were shorter, often having branched appearance and peculiar kinky shape. Also, callose plugs were rare or absent in mutant pollen tubes, compared to the wild type. Despite morphological aberrations, twisted pollen tube appeared to have same ultrastructural features as

745:6809:AT5G54690	irx8-4	Collapsed xylem vessels in roots and stems, suggesting a decrease in xylem wall strength in the homozygous mutants.	17237350
746:6814:AT5G54690	irx8-4	Where in wild-type roots xylem files form a closely packed array of cell  files, leading to the regular occurrence of  tripartite cell corners between xylem cell files, the central cylinder of homozygous mutant roots was disorganized and  the xylem files did not form a closely packed array. As a  consequence, tripartite xylem cell  corners could not be detected in the mutant.	17237350
747:6816:AT5G54690	irx8-5	Collapsed xylem vessels in roots and stems, suggesting a decrease in xylem wall strength in the homozygous mutants.	17237350
748:6821:AT5G54690	irx8-5	Where in wild-type roots xylem files form a closely packed array of cell  files, leading to the regular occurrence of  tripartite cell corners between xylem cell files, the central cylinder of homozygous mutant roots was disorganized and  the xylem files did

767:6947:AT5G60920	CS8541	When grown on media containing 0.5% sucrose, the mutant had reduced root length but no apparent radial cell expansion.	7743935
768:6950:AT5G61070	SALK_006938	Increased root hair density.	16176989
769:6974:AT5G62165	agl42-4	No root anatomy phenotype	15937229
770:6980:AT5G62500	eb1b-1	Plants grown on agar plates have roots that skew towards the left occasionally forming clockwise loops with twisted epidermal cell files. Also has reduced/delayed response to gravity.	18281505
771:6983:AT5G62940	hca2	Plants are dwarfed. In the hca2 mutant inflorescence stems, petioles,and main veins of leaves, the ordered patterning of vascular bundles is replaced by a continuous vascular cambium undergoing periclinal divisions to produce radial files of xylem and phloem.Interfascicular cambium is extensively formed/developed between the vascular bundles, and the area normally occupied by the interfascicular parenchyma cells is replaced by vascular tissues.  Continuous phloem tissu

As we can see, the output creates a new list starting with the number 1 that corresponds to the first Locus code with a phenotype where the word root is included. However, it also displays a lot information we don't want to know right now. To eliminate it, I added a -o in the last grep command so now it will only display the Locus codes (the list will start with the number 1 again). To rewrite all of them in a new file I typed > Root-associated-Loci.txt and cat Root-associated-Loci.txt to check if it worked properly.      

In [9]:
grep -n '\root' Locus_Germplasm_Phenotype_20130122.txt | grep -E -ino '\AT[0-9]G[0-9]{5}' > Root-associated-Loci.txt
cat Root-associated-Loci.txt

1:AT1G01550
2:AT1G01550
3:AT1G01550
4:AT1G02970
5:AT1G03000
6:AT1G03060
7:AT1G03060
8:AT1G03060
9:AT1G03060
10:AT1G04020
11:AT1G04030
12:AT1G04160
13:AT1G04160
14:AT1G04160
15:AT1G04220
16:AT1G04250
17:AT1G05180
18:AT1G05180
19:AT1G05180
20:AT1G05850
21:AT1G05850
22:AT1G05850
23:AT1G06290
24:AT1G06290
25:AT1G08030
26:AT1G08260
27:AT1G08370
28:AT1G08430
29:AT1G08430
30:AT1G09100
31:AT1G09110
32:AT1G09940
33:AT1G10130
34:AT1G10570
35:AT1G11130
36:AT1G12240
37:AT1G12610
38:AT1G13290
39:AT1G14290
40:AT1G14660
41:AT1G14660
42:AT1G15550
43:AT1G15690
44:AT1G16150
45:AT1G17110
46:AT1G17170
47:AT1G17580
48:AT1G17580
49:AT1G18350
50:AT1G18800
51:AT1G19220
52:AT1G19220
53:AT1G19220
54:AT1G19220
55:AT1G19220
56:AT1G19220
57:AT1G19220
58:AT1G19220
59:AT1G19220
60:AT1G19220
61:AT1G19220
62:AT1G19220
63:AT1G19220
64:AT1G20700
65:AT1G21270
66:AT1G21700
67:AT1G22920
68:AT1G22920
69:AT1G23010
70:AT1G23010
71:AT1G23310
72:AT1G23320
73:AT1G23320
74:AT1G26870
75:AT1G27320
76:AT1G27320
77:AT1G27320
78:AT1G2

591:AT5G09810
592:AT5G10250
593:AT5G10330
594:AT5G10510
595:AT5G10510
596:AT5G11260
597:AT5G11260
598:AT5G11260
599:AT5G11260
600:AT5G11260
601:AT5G11260
602:AT5G11260
603:AT5G13290
604:AT5G13570
605:AT5G13570
606:AT5G13570
607:AT5G13570
608:AT5G13570
609:AT5G14420
610:AT5G14420
611:AT5G16530
612:AT5G16780
613:AT5G18170
614:AT5G18560
615:AT5G19040
616:AT5G19040
617:AT5G19040
618:AT5G19040
619:AT5G19040
620:AT5G19040
621:AT5G19660
622:AT5G20410
623:AT5G20490
624:AT5G20490
625:AT5G20490
626:AT5G20490
627:AT5G20490
628:AT5G20490
629:AT5G20490
630:AT5G20730
631:AT5G20730
632:AT5G20730
633:AT5G20730
634:AT5G20730
635:AT5G20730
636:AT5G20730
637:AT5G20730
638:AT5G20730
639:AT5G20730
640:AT5G20730
641:AT5G20730
642:AT5G20730
643:AT5G20730
644:AT5G20830
645:AT5G20830
646:AT5G20830
647:AT5G23720
648:AT5G24270
649:AT5G24270
650:AT5G24270
651:AT5G24310
652:AT5G24520
653:AT5G25220
654:AT5G25350
655:AT5G25350
656:AT5G25350
657:AT5G25350
658:AT5G25350
659:AT5G26030
660:AT5G27320
661:AT5G27320
662:AT

Problem 8
As I did in the last problem, firstly I used the command of Problem 6 (excluding the -o) followed by a pipe and the command grep -E -in '\s[0-9]{4,30}' to select only the Pubmed ids and dismiss other combination of numbers in the rest of columns.

In [10]:
grep -n '\root' Locus_Germplasm_Phenotype_20130122.txt | grep -E -in '\s[0-9]{4,30}'

1:23:AT1G01550	bps1-2	Growth of bps1-2 mutants on CPTA-supplemented medium resulted in partial rescue of both leaf and root defects. bps1-2 mutants grown on control growth medium have small radialized leaves with very little vascular tissue, and very short misshapen knotted-looking roots. By contrast, bps1-2 mutants grown on CPTA-supplemented medium produced larger flattened leaves that contained primary and secondary veins, and smooth elongated roots.	17217459
2:24:AT1G01550	bps1-2	Mutant roots were abnormal; primary and lateral roots were short, root hairs formed close to the root apex, and root defects were most severe when plants were grown at low temperature.	15458645
3:26:AT1G01550	bps1-2	Under high light conditions (approximately 200 &mu;E.m-2.sec-1), CPTA-treated seedlings were completely photobleached and mutants showed rescue of both leaf and root development. Under moderate (approximately 100 &mu;E.m-2.sec-1) and low (30 &mu;E.m-2.sec-1) light intensities, seedlings are inco

26:253:AT1G08260	tilted1-4	Twenty-five percent of embryos show abnormal cell divisions during embryogenesis-large cells- decreased rate of cell division. Division of the hypohyseal cell abnormal leading to abnormal placement of the root pole. Homozygotes show delayed flowering, abnormal floraly phyllotaxy and abnormal ovules.	16278345
27:255:AT1G08370	dcp1-2	The homozygous progeny is seedling lethal, showed arrested postembryonic development including cotyledon expansion, development of vascular networks, root elongation, and shoot development.	17485080
28:256:AT1G08430	Atalm1-KO	On Al3+ medium, the mutant has significantly shorter roots than the Col wild type. In the absence of aluminum stress, the roots of mutant and wild type grew similarly.	16740662
30:289:AT1G09100	rpt5b	Insensitive to high-glucose repression of cotyledon expansion, chlorophyll accumulation, true-leaf development and root elongation. The response is specific to glucose but not to osmotic changes.	17081979
31:294:A

56:572:AT1G19220	CS24626	The phenotype of the double mutant is most obvious at its seedling stage, with its most prominent phenotype being severely impaired  lateral root formation. Its primary roots fail to produce lateral roots in 2-week-old seedlings. However, double mutant seedlings start to generate several lateral roots after ~2 weeks of growth, and their morphological appearance is normal.	15659631
57:574:AT1G19220	CS24626	nph4-1 arf19 double mutant;  agravitropic response in both hypocotyls and roots; impaired phototropic response in hypocotyls; impaired lateral root formation; small plant size; small and epinastic rosette leaves; reduced auxin sensitivity	15659631
58:577:AT1G19220	CS24627	The phenotype of the double mutant is most obvious at its seedling stage, with its most prominent phenotype being severely impaired  lateral root formation. Its primary roots fail to produce lateral roots in 2-week-old seedlings. However, double mutant seedlings start to generate several late

92:882:AT1G27320	stip-d ahk2-2 ahk3-3 cre1-12	Small SAM and limited root growth.	20110319
93:894:AT1G27760	atsat32	Reduced fertility. Increased root length (longer roots). Hypersensitive to salt stress.	19210750
94:896:AT1G28300	CS3868	Embryo lethal, embryos not viviparous, cotyledons partially transformed into leaves, complex vascular pattern observed in cotyledons. Embryos are sensitive to abscisic acid. Mutant seeds were usually desiccation tolerant at maturity, although their viability was reduced slightly after several weeks of storage. Dry seeds often germinated to produce seedlings with distorted cotyledons. This defect resulted from degeneration of cotyledon tips prior to germination. Some trichomes form on cotyledons after germination and stomata are found on surface of cotyledons of mutant embryos The cotyledon surface on seedlings produced from dry seeds lacked trichomes (likely because any trichomes initiated during embryogenesis did not survive desiccation). Mutant seeds w

119:1124:AT1G49240	act2-1/act8-2	Lacks root hairs.Root hairs initiate but do not develop further.	19304937
120:1125:AT1G49240	act8-2	Short root hairs. 50% reduction in root hair length compared to wild type.	19304937
121:1126:AT1G49240	act8-2/act7-4	Plants are dwarf. Leaves are smaller and contain fewer cells and fewer lobes than wild type. Reduced number of trichome branches. Petals are narrow. Roots are more severely affected and produce short root hairs. Cell files are irregular and the elongation zone is much reduced.	19304937
122:1162:AT1G50960	ga2ox7-2	Longer roots when plants are grown under salt stress (50-100nM NaCl).	18643985
123:1163:AT1G51190	plt3-1/plt2-2/plt1-4	Rootless.Embryonic root differentiates 3 days after germination. Adventitious roots arrest 6 days after germination.	17960244
124:1199:AT1G52340	aba2-2	Mutant showed a similar sensitivity to 25 mM trehalose as wild type (inhibition of root growth).	17031512
125:1234:AT1G53330	CB_1265	30-40% T3 seeds are embryo leth

148:1316:AT1G58440	dry2	Plants are hypersensitive to drought. Mutants have decreased stomatal conductance when grown in low relative humidity. Root architecture is affected: roots are 60% shorter than wild type and produce more lateral roots. Lateral roots are shortr and branched.Plants are shorter and leaves are yellow green. Chemically, the mutants have decreased sterols in the roots, decreased chlorophyll levels and increased levels of proline.	19309460
149:1322:AT1G59870	ko-1	shoot FW, root length and chlorophyll contents of ko-1 plants are significantly lower than those of wild-type plants	17355438
150:1323:AT1G59870	ko-2	shoot FW, root length and chlorophyll contents of ko-2 plants are significantly lower than those of wild-type plants	17355438
151:1328:AT1G60220	ots1-1 ots2-1	The ots1-1 ots2-1 double mutant plants flower earlier than wild type plants under long days and short days. They do not differ notably from wild type plants with respect to their overall growth rate, inflor

180:1636:AT1G70070	ise2-1	Homozygous mutant seeds can germinate in vitro on agar plates containing Murashige and Skoog (MS) salt plus 1% sucrose, although the  growth/developmental rate of the homozygous seedlings was slower compared with that of wild-type seedlings. Whereas wild-type seedlings displayed fully expanded green cotyledons, true leaves and a well-developed root system by 14 d after germination, homozygous seedlings were much smaller and white or pale green in color. Underdeveloped cotyledons emerged from the homozygous seeds but expanded very slowly and sometimes showed bumpy irregular surfaces and occasional anthocyanin expression. Partial greening was observed in ise2 seedlings only after culture for several weeks. Most of the homozygous seedlings developed leaf-like structures that were translucent and contained trichomes. The aerial portions of the homozygous seedlings generally were stunted and thicker and frequently exhibited callus-like structures after extended per

199:1795:AT1G78240	tsd2-1	Increased activity of axial meristems, reduced root growth and enhanced de-etiolation.	17461780
200:1833:AT1G79580	SALK_143526C	Additional columella root cap and lateral root cap cell layers in the mature embryo and postembryonic root development. Root meristem length and cell number is similar to WT. Extra layer of small, stem-cell-like cells, below the columella and lateral root cap stem cells.	19081078
201:1834:AT1G79580	SALK_143526C	Delayed maturation and shedding of lateral root cap cells. The root cap extends into the differentiation zone.	20197506
203:1844:AT1G80100	ahp6-1	increased number of vascular cell files with  intervening procambial and phloem cell files; protoxylem differentiation  occurred sporadically along the root	16400151
204:1845:AT1G80100	ahp6-2	increased number of vascular cell files with intervening procambial and phloem cell files; protoxylem differentiation occurred sporadically along the root	16400151
205:1855:AT1G80340	ga3ox1-3 ga3

255:2106:AT2G03680	CS6547	Defective in directional cell elongation processes; abnormal cortical microtubule function; exhibits right-handed helical growth in roots and etiolated hypocotyls; epidermal cell files of roots are twisted to form right-handed helices; on vertically oriented hard agar plates, roots grow to the right when viewed from above the agar plates; this skewed root growth is driven by the friction between agar surface and helical epidermal cell files; phenotype is enhanced under the conditions that accelerate cell elongation, under such conditions, epidermal cells undergo isotropic cell expansion, resulting in spherically shaped cells protruding from the organ surface.	15084720
256:2110:AT2G03680	spr1-6	Mutant plants exhibit altered patterns of root and organ growth as a result of defective anisotropic cell expansion. Roots, etiolated hypocotyls, and leaf petioles exhibit right-handed axial twisting, and root growth on inclined agar media is strongly right skewed. Corti

280:2386:AT2G25490	ebf1-1 ebf2-1	When germinated in the dark, the mutant display a ctr1-like phenotype (short hypocotyls, short roots, and exaggerated apical hooks).	14675533
281:2387:AT2G25490	ebf1-1 ebf2-1	When grown in the light, the double mutants display slightly longer hypocotyls, curled unexpanded cotyledons, and short and thicker roots with ectopic root hairs, similar to the ctr1 mutant. Moreover, the average area of the leaf epidermal cells was reduced 5-fold compared to wild type.	14675533
282:2389:AT2G25490	ebf1-2 ebf2-1	ctr1-like phenotype (short hypocotyls, short roots, and exaggerated apical hooks).	14675533
283:2391:AT2G25490	ebf1-3 ebf2-3 eil1-3	Modest rescue of the ebf1-3 ebf2-3 phenotype. Seedlings continue grow and develop at a slow pace after germination. Cotyledons partially expanded and turned pale green. Stunted root densely covered with root hair. Plants never flowered or set seeds.	17307926
284:2394:AT2G25680	CS859718	molybdate concentrations in shoot and root 

311:2732:AT2G36490	ror1-1 ros1-1	Increased sensitivity to DNA damage,decreased gene silencing as measured by expression of reporter genes, small plants, decreased trichome number, early flowering in long days,decreased cell division, short roots.	16326925
312:2777:AT2G38120	CS8040	auxin resistant; narrow, irregular rosette leaves, slightly curled around leaf axis; roots elongate on auxin-containing medium; defective root gravitropism; reduced number of lateral roots - greater reduction than for either single mutant; dwarf, bushy plants; reduced plant height; ethylene resistant; reduced fertility.	7704045
313:2788:AT2G38120	aux1-113	Partial root gravitropic phenotype.	15486104
314:2789:AT2G38120	aux1-114	Partial root gravitropic phenotype.	15486104
315:2791:AT2G38120	aux1-116	Partial root gravitropic phenotype.	15486104
316:2798:AT2G38120	axr4-1 aux1-7	auxin resistant; narrow, irregular rosette leaves, slightly curled around leaf axis; roots elongate on auxin-containing medium; defectiv

339:3005:AT2G44950	hub1-4	Mutant growth was defective compared with  that of wild type. The mutants were reduced in size and had pale-green laminas with an  irregular blade surface and thinner inflorescence stems, but root growth was not  visibly reduced. Furthermore, the fresh weight of mutant was approximately half of that of wild type.	17329565
341:3090:AT2G46240	SALK_009534	Mutant plants exhibited earlier flowering and shorter vegetative and reproductive phases, producing more branched roots and inflorescences compared to wild-type plants. Senescence occurred earlier in mutants than the wild-type plants.	16636050
342:3092:AT2G46240	SALK_058290	Mutant plants exhibited earlier flowering and shorter vegetative and reproductive phases, producing more branched roots and inflorescences compared to wild-type plants. Senescence occurred earlier in mutants than the wild-type plants.	16636050
343:3094:AT2G46370	CS8072	JA-Ile levels are seven-fold lower in the jar1-1 mutant compared to wild t

371:3318:AT3G06810	ibr3-1	Roots moderately resistant to elongation inhibition by 2,4-DB (2,4-Dichlorophenoxybutyric acid) but remained sensitive to inhibition by 2,4-D (2,4-Dichlorophenoxyacetic acid). Fewer lateral roots are produced without induction which becomes more pronounced in response to IBA (Indole-3-butyric acid) treatment.	17277896
372:3319:AT3G06810	ibr3-11	Roots moderately resistant to elongation inhibition by 2,4-DB (2,4-Dichlorophenoxybutyric acid) but remained sensitive to inhibition by 2,4-D (2,4-Dichlorophenoxyacetic acid). Fewer lateral roots are produced without induction which becomes more pronounced in response to IBA (Indole-3-butyric acid) treatment.	17277896
373:3320:AT3G06810	ibr3-2	Roots moderately resistant to elongation inhibition by 2,4-DB (2,4-Dichlorophenoxybutyric acid) but remained sensitive to inhibition by 2,4-D (2,4-Dichlorophenoxyacetic acid). Fewer lateral roots are produced without induction which becomes more pronounced in response to IBA (Indo

394:3527:AT3G14370	wag2-1	Morphologically similar to wild type. When seedlings were grown on vertical plates, root exhibited wavy pattern.	16460509
395:3542:AT3G16150	CS67896	Increased sensitivity to delay in root hair emergence in response to exogenous asparagine. Increased accumulation of asparagine in root when grown on medium containing asparagine as a single nitrogen source. Increased seed size under conditions of nitrogen deficiency. Increased concentration of free asparagine and other free amino acids in seed when grown under low illumination.	21800258
396:3543:AT3G16150	CS67896	increased sensitivity to inhibition of root elongation by exogenous asparagine	21800258
397:3544:AT3G16150	SALK_082112	increased sensitivity to inhibition of root elongation by exogenous asparagine	21800258
398:3545:AT3G16150	WiscDsLox382A10	increased sensitivity to inhibition of root elongation by exogenous asparagine	21800258
399:3548:AT3G16785	pldzeta1 pldzeta2	Under limited phosphate, primary root el

424:3721:AT3G23630	atipt1 atipt 3 atipt 5 atipt7	Relative to wild type, both the number of lateral roots (longer than 1cm) and the total length of the lateral roots are increased.	17062755
425:3738:AT3G23630	atipt3 atipt 5 atipt 7	Enhanced root growth rate resulting in longer roots.	17363254
426:3741:AT3G23630	atipt3 atipt 5 atipt 7	Mutant root meristems showed an increased number of meristematic cells already at two days after germination (dpg) and accumulated cells also after five dpg, vastly exceeding the fixed number of cells of wild-type meristems.	17363254
427:3742:AT3G23630	atipt3 atipt 5 atipt 7	Relative to wild type, both the number of lateral roots (longer than 1cm) and the total length of the lateral roots are increased.	17062755
428:3785:AT3G25520	oli5-1	Shorter primary roots with respect to the wild type.	19392710
430:3789:AT3G26090	CS6537	Altered cell proliferation; increased cell elongation in hypocotyls in darkness (longer hypocotyls than wild type); increased cell prod

457:4074:AT3G51780	SALK_033845	Mutant plants exhibited earlier flowering and shorter vegetative and reproductive phases, producing more branched roots and inflorescences compared to wild-type plants. Senescence occurred earlier in mutants than the wild-type plants.	16636050
458:4109:AT3G53020	CS11180	Defects in apical-basal gynoecium patterning (similar to ett and mp mutants). Mutant plant gynoecia had shorter ovaries and longer gynophores than the wild type, but the total length of the gynoecium is normal before pollination. After pollination, the gynoecium of failed to elongate, resulting in shorter siliques which contained arrested seeds, resulting in lower fertility. The stv1-1 ovules had shorter integuments than wild-type ones, and in some extreme cases, the gametophyte was protruding from the integuments. Mutant plants had defects in vascular system and embryo organization, which are also associated with auxin signaling. Most cotyledons in mutant plants had abnormal vascular patt

467:4274:AT3G59420	acr4-2	Increased number of lateral root meristems initate.	18948541
468:4275:AT3G59420	acr4-2/ccr1-3	Increased number of lateral root meristems, greater than acr4 single mutant alone.	18948541
469:4281:AT3G59770	sac9-1	slow growing, small leaves with short petioles, accumulates anthocyanin in the leaves, small and constitutively closed guard cells, shorter primary roots, fewer lateral roots, hypersensitive to osmotic stress, cold stress and high-light stress	15923324
470:4282:AT3G59770	sac9-2	slow growing, small leaves with short petioles, accumulates anthocyanin in the leaves, small and constitutively closed guard cells, shorter primary roots, fewer lateral roots, hypersensitive to osmotic stress, cold stress and high-light stress	15923324
471:4305:AT3G61650	TUBG1-1/tubg1-1 TUBG2-2/tubg2-2	During early seedling growth, cotyledons were small and distorted, and only two leaf primordia were inittaited at the shoot apex. Dome-shape soot apical meristem was distorted. Ro

498:4558:AT4G10310	athtk1-3	athtk1-3 mutants have no obvious phenotypic abnormalities when grown in typical greenhouse conditions. But, across a range of different hydroponically-provided sodium concentrations, these mutants accumulate less sodium in their roots than wild type plants and more sodium in their leaves than wild type plants. Accordingly, the roots of these mutants are less sensitive to salt in root elongation assays, and the leaves are hypersensitive to salt in plant growth assays.	12417304
499:4562:AT4G10350	brn1-1; brn2-1	Columella root cap cells fail to detach.	20197506
501:4567:AT4G10380	SALK_122287C	Inhibition root, shoot, flowers and siliques growth when Boron was supplied at concentrations of <10 uM.	16679457
502:4618:AT4G14550	slr-1	Absence of lateral roots. Defective response to auxin stimulus in the root.	11862947
503:4639:AT4G14850	loi1-1	loi1-1 mutants are less sensitive to inhibitors of the MEP and MVA pathways of isoprenoid biosynthesis than wild type plants.

525:4961:AT4G29040	hlr-1	The hlr-1 (halted root-1) mutant has defective cell organization in the post-embryonic root and shoot apical meristems. Quiescent center cell identity in the root is lost after germination and expression of the organizing center marker WUSCHEL in the shoot is disrupted. However, cells in differentiated tissues maintain their normal identities. hlr-1 mutants have shorter roots, aberrant lateral root primordia positioning, abnormal leaf shape and phyllotaxy, and irregular shoot structure.	15073153
526:4962:AT4G29040	hlr-2	The hlr-2 mutant is reported to share the same morphological characteristics with the hlr-1 mutant, but to have a slightly milder phenotype. hlr-1 has defects in root and shoot apical meristem organization, and root and shoot development.	15073153
527:4963:AT4G29040	hlr-2	The rpt2a-2 mutant has modest defects in ubiquitin-mediated proteolysis via the 26S proteasome, rendering it somewhat more sensitive to misfolded proteins that are degraded in 

552:5264:AT4G38130	hda19-1	Temperature sensitive phenotype.  When grown at 24 degrees C, seedlings have narrow cotyledons compared to wt.  When grown at 29 degrees C, seedlings display several tpl-1-like phenotypes, including monocots, tubes and pins.  Both shott and root can be disorganized.	16763149
553:5265:AT4G38240	SALK_073650	strongly repressed root growth when exposed to salt or manitol, callose accumulation at root tips exposed to salt.	18408158
554:5275:AT4G39350	ixr2-1 cesa2	The cesa2/cesa6 double mutant has reduced hypocotyl length in the dark, reduced root length in the light, strong dwarf and bushy.	17878303
555:5276:AT4G39350	prc1-1 cesa2-1	The double mutant seedlings had significantly shorter and swollen roots. The plants had stunted growth and reduced size.	17878302
556:5325:AT5G01180	atptr1-2 atptr5-1	The ptr1-2 ptr5-2 double mutant grows normally on soil and on AM medium. Its pollen tubes germinate and elongate normally on germination media. The tubes can germinate no

575:5462:AT5G03730	ctr1-2	Mutation had dramatic effects on the morphology of adult plants. The rosette leaves of mutant plants were much smaller than wild-type leaves. The mutants bolted - l-2 weeks later, the early flowers were infertile, the root system was much  less extensive, and the inflorescence was much smaller than those of the wild type.  The gynoecium of the mutant elongated significantly earlier relative to the rest of the developing flower, often protruding  out of the unopened buds.	8431946
576:5463:AT5G03730	ctr1-3	Mutation had dramatic effects on the morphology of adult plants. The rosette leaves of mutant plants were much smaller than wild-type leaves. The mutants bolted - l-2 weeks later, the early flowers were infertile, the root system was much  less extensive, and the inflorescence was much smaller than those of the wild type.  The gynoecium of the mutant elongated significantly earlier relative to the rest of the developing flower, often protruding  out of the uno

598:5686:AT5G11260	SALK_096651C	Decreased sensitivity to ABA induced lateral root inhibition. Increased sensitivity to osmotic and salt stress. Increased ABA tolerance.	18065688
599:5688:AT5G11260	hy5-215	Mutants grown on agar plates have widely  spread lateral roots. Compared with that of wild-type plants, the direction of lateral root growth is nearly horizontal rather than downward, indicating alteration of the gravitropic response. The angles between the lateral roots and the orientation of gravity are larger than those of the wild type. In addition, secondary lateral roots grow and change their direction of elongation to the horizontal.	9367981
600:5693:AT5G11260	hy5-KS50	At 4 DAG, the number of lateral root primordia is increased in the Wassilewskija (Ws) wild type , but the increase was two times greater in the hy5-Ks50 mutant than in Ws.	9367981
601:5695:AT5G11260	hy5-KS50	Mutants grown on agar plates have widely  spread lateral roots. Compared with that of wild-type plants, th

624:5973:AT5G20490	SALK_067972C	There are no obvious defects in the aerial tissues of xi-k mutant plants grown under normal conditions. The root length of the mutant is also normal, but, it has shorter roots hairs than wild-type plants. There are decreased rates of movement of Golgi stacks, peroxisomes, and mitochondria in root hairs and leaves.	18178669
625:5974:AT5G20490	SALK_136682C	The xik-1 mutant has normal root growth, but its root hairs are only 49% of the length of wild type root hairs, and they have a hooked shape when grown on media with sugar. The stem trichomes are crooked and wavy and often have abnormally elongated stalks. The trichomes on the leaves also have size and shape irregularities. But, the total number of trichomes is the same for mutants and wild-type plants.	17458634
626:5975:AT5G20490	WiscDsLox417C12 (HOMOZYGOUS)	There are no obvious defects in the aerial tissues of xi-k mutant plants grown under normal conditions. The root length of the mutant is also norma

647:6157:AT5G23720	phs1-1	phs1-1 exhibits phenotypes indicative of compromised cortical microtubule functions, such as left-handed helical growth of seedling roots, defective anisotropic growth at low doses of microtubule-destabilizing drugs, enhancement of the temperature-sensitive microtubule organization1-1 phenotype, and less ordered and more fragmented cortical microtubule arrays compared with the wild type.	15208393
648:6164:AT5G24270	sos3-1 siz1-1	Decrease of macronutrients concentration in the MS salt formulation to 1/20X results in a substantial reduction in primary root growth.	15894620
649:6165:AT5G24270	sos3-1 siz1-1	Mutant grown in greenhouse or growth chamber exhibits reduced shoot and root biomass relative to the wild-type.	15894620
650:6167:AT5G24270	sos3-1 siz1-1	Substantially more pronounced prototypical Pi starvation root architecture responses than sos3-1 or wild-type seedlings. On Pi-limited medium, mutant seedlings exhibit an inhibition of primary root growth, an 

691:6389:AT5G39510	sgr4-1	Inflorescence stems and hypocotyls of mutant plants showed abnormal gravitropic response, while root gravitropism was normal.	9210330
692:6394:AT5G39610	nac2-1	Not significantly affected in lateral root development.	16359384
694:6466:AT5G43900	SALK_055785C	There are no obvious defects in the aerial tissues of xi-2 mutant plants grown under normal conditions. The root length of the mutant is also normal, but, it has shorter roots hairs than wild-type plants.	18178669
695:6467:AT5G43900	xi-2 / xi-b	The overall growth of the xi-2 / xi-b double mutant is normal. In the leaf epidermis, Golgi stacks and peroxisomes move more slowly in these mutant cells than in wild-type cells. The rate of mitochondrial movement is also slightly reduced. Root hair length is reduced to ~15% of the wild type root hair length in these double mutants, but their root hair density is very similar to the wild type density.	19060218
696:6468:AT5G43900	xi-k / xi-2	The overall growth of the x

717:6592:AT5G48150	pat1-1	No aberrant root phenotype and no resistance to paclobutrazol (1-100 &mu;M).	10817761
718:6596:AT5G48160	obe1-1 obe2-1	Premature termination and loss of the shoot apical meristem. Failure to establish or maintain the root apical meristem.  Aberrant embryonic development.	18403411
719:6597:AT5G48160	obe1-1 obe2-2	absence of roots and defective development of the vasculature.	19392692
720:6602:AT5G48300	aps1-SALK_040155	In short days plants have delayed growth and were late flowering.In long day conditions, plants also had delayed growth but flowering time was unaffected. Lacks ADP-Glc PPase activity in leaves and no starch can be found in leaves or roots.	18614708
721:6618:AT5G49190	GABI_377G03	The amounts of glucose, fructose, sucrose, cellulose (roots) and starch in the mutant were not statistically significantly different  from those of the equivalent wild-type lines grown under the same conditions at the same time.	17257168
722:6620:AT5G49190	sus2 (SALK_076

745:6809:AT5G54690	irx8-4	Collapsed xylem vessels in roots and stems, suggesting a decrease in xylem wall strength in the homozygous mutants.	17237350
746:6814:AT5G54690	irx8-4	Where in wild-type roots xylem files form a closely packed array of cell  files, leading to the regular occurrence of  tripartite cell corners between xylem cell files, the central cylinder of homozygous mutant roots was disorganized and  the xylem files did not form a closely packed array. As a  consequence, tripartite xylem cell  corners could not be detected in the mutant.	17237350
747:6816:AT5G54690	irx8-5	Collapsed xylem vessels in roots and stems, suggesting a decrease in xylem wall strength in the homozygous mutants.	17237350
748:6821:AT5G54690	irx8-5	Where in wild-type roots xylem files form a closely packed array of cell  files, leading to the regular occurrence of  tripartite cell corners between xylem cell files, the central cylinder of homozygous mutant roots was disorganized and  the xylem files did

771:6983:AT5G62940	hca2	Plants are dwarfed. In the hca2 mutant inflorescence stems, petioles,and main veins of leaves, the ordered patterning of vascular bundles is replaced by a continuous vascular cambium undergoing periclinal divisions to produce radial files of xylem and phloem.Interfascicular cambium is extensively formed/developed between the vascular bundles, and the area normally occupied by the interfascicular parenchyma cells is replaced by vascular tissues.  Continuous phloem tissue consisting of several layers of sieve tubes and companion cells encircles the vascular cambium in hca2.  The vascular tissue in the  hypocotyls and roots of hca2 seedlings exhibit a WT phenotye.	19915089
772:6985:AT5G62950	hca2	Plants are dwarfed. In the hca2 mutant inflorescence stems, petioles,and main veins of leaves, the ordered patterning of vascular bundles is replaced by a continuous vascular cambium undergoing periclinal divisions to produce radial files of xylem and phloem.Interfascicula

798:7195:MOP2	mop2-1	deficiencies in post-transcriptional regulation of PIN genes. Defects in auxin-controlled growth responses as well as root and shoot development	17651372
799:7202:RIB1	rib1	Seedlings have shorter primary roots and increased numbers of lateral roots compared to Nossen wild type.  Adults were indistinguishable from wild type in terms of shoot length and secondary inflorescence formation.  Hypocotyl and root IBA basipetal transport are decreased.  Root acropetal IBA transport is increased.	16258013
800:7203:SGR6	sgr6-1	Inflorescence stems of mutant plants showed abnormal gravitropic response, while gravitropism in hypocotyls and roots was not altered.	9210330


Again, after the list created by the output, I added a -o to the second grep command and I typed > Root-associated-Publications.txt to rewrite it in a different file and cat Root-associated-Publications.txt to check that everything worked properly. 

In [11]:
grep -n '\root' Locus_Germplasm_Phenotype_20130122.txt | grep -E -ino '\s[0-9]{4,30}' > Root-associated-Publications.txt
cat Root-associated-Publications.txt

1:	17217459
2:	15458645
3:	17217459
4:	17209125
5:	14745029
6:	19392685
7:	19392685
8:	19392685
9:	19392685
10:	18591352
11:	18591352
12:	18178669
13:	18178669
14:	19060218
15:	18786002
16:	16367964
17:	17655650
18:	17655650
19:	7704045
20:	7743935
21:	7743935
22:	7743935
23:	15743450
24:	15743450
25:	19666544
26:	16278345
27:	17485080
28:	16740662
30:	17081979
31:	17081979
33:	18024560
34:	18849491
35:	15618487
36:	16481625
37:	18643985
38:	18643975
39:	18612100
40:	17270011
41:	17270011
42:	16460513
43:	16210544
44:	16286448
45:	18485060
47:	18178669
48:	19060218
49:	16377756
51:	15659631
52:	15659631
53:	15659631
54:	15659631
55:	15659631
56:	15659631
57:	15659631
58:	15659631
59:	15659631
60:	15659631
61:	15659631
62:	16461383
63:	16461383
64:	18950478
65:	16623892
67:	17307927
68:	17307927
69:	17496893
70:	17496893
71:	17318317
72:	18394997
73:	18394997
74:	19081078
75:	16357038
76:	16357038
77:	16357038
78:	16357038
79:	11479382
80:	11479382
84:	16361392
85:	16361392
86:	17363254

664:	17319849
665:	16183846
666:	16357038
667:	16357038
668:	16357038
669:	16357038
670:	11479382
671:	11479382
674:	15166290
677:	16361392
678:	16361392
679:	16361392
686:	20110319
687:	17257168
688:	17257168
689:	17257168
690:	18768907
691:	9210330
692:	16359384
694:	18178669
695:	19060218
696:	19060218
697:	18326829
698:	18326829
699:	7770517
700:	7770517
701:	7770517
702:	7770517
703:	18494853
704:	15908599
705:	15908599
706:	10911994
707:	10911994
708:	8819871
710:	18039663
711:	19168677
712:	19168677
713:	18356490
714:	18356490
715:	18356490
716:	18356490
717:	10817761
718:	18403411
719:	19392692
720:	18614708
721:	17257168
722:	17257168
723:	17257168
724:	15863435
725:	14675453
726:	14675453
727:	14675453
728:	14675453
729:	17873088
730:	17873088
731:	16415212
734:	17540573
735:	20609115
736:	17237350
737:	17322407
738:	17237350
739:	17237350
740:	17237350
741:	17237350
742:	17237350
743:	17237350
744:	17237350
745:	17237350
746:	17237350
747:	17237350
748:	17237350
749:	1732240

Problem 9
In this problem I modified the grep command I used in problem 7, to isolate and organise all the Locus codes associated with the roots by the chromosome where we can find that particular locus. To achieve that I rewrited all of them into new files depending on their location, so the ones in the chromosome 1 will be included in a file called Chromosome1.txt, the ones in the chromosome 2 will be in a file called Chromosome2.txt etc. In addition, I included the command wc -l to count the number of lines of the new files so I can add them together and compare the result with the line count of the original document, Root-associated-Loci.txt.  

In [12]:
grep -E -ino '\AT1G[0-9]{5}' Root-associated-Loci.txt > Chromosome1.txt
wc -l Chromosome1.txt
cat Chromosome1.txt

206 Chromosome1.txt
1:AT1G01550
2:AT1G01550
3:AT1G01550
4:AT1G02970
5:AT1G03000
6:AT1G03060
7:AT1G03060
8:AT1G03060
9:AT1G03060
10:AT1G04020
11:AT1G04030
12:AT1G04160
13:AT1G04160
14:AT1G04160
15:AT1G04220
16:AT1G04250
17:AT1G05180
18:AT1G05180
19:AT1G05180
20:AT1G05850
21:AT1G05850
22:AT1G05850
23:AT1G06290
24:AT1G06290
25:AT1G08030
26:AT1G08260
27:AT1G08370
28:AT1G08430
29:AT1G08430
30:AT1G09100
31:AT1G09110
32:AT1G09940
33:AT1G10130
34:AT1G10570
35:AT1G11130
36:AT1G12240
37:AT1G12610
38:AT1G13290
39:AT1G14290
40:AT1G14660
41:AT1G14660
42:AT1G15550
43:AT1G15690
44:AT1G16150
45:AT1G17110
46:AT1G17170
47:AT1G17580
48:AT1G17580
49:AT1G18350
50:AT1G18800
51:AT1G19220
52:AT1G19220
53:AT1G19220
54:AT1G19220
55:AT1G19220
56:AT1G19220
57:AT1G19220
58:AT1G19220
59:AT1G19220
60:AT1G19220
61:AT1G19220
62:AT1G19220
63:AT1G19220
64:AT1G20700
65:AT1G21270
66:AT1G21700
67:AT1G22920
68:AT1G22920
69:AT1G23010
70:AT1G23010
71:AT1G23310
72:AT1G23320
73:AT1G23320
74:AT1G26870
75:AT1G27320
76:AT1G27320
7

In [13]:
grep -E -ino '\AT2G[0-9]{5}' Root-associated-Loci.txt > Chromosome2.txt
wc -l Chromosome2.txt
cat Chromosome2.txt

151 Chromosome2.txt
206:AT2G01830
207:AT2G01830
208:AT2G01830
209:AT2G01830
210:AT2G01830
211:AT2G01830
212:AT2G01830
213:AT2G01830
214:AT2G01830
215:AT2G01830
216:AT2G01830
217:AT2G01830
218:AT2G01830
219:AT2G01830
220:AT2G01830
221:AT2G01830
222:AT2G01830
223:AT2G01830
224:AT2G01830
225:AT2G01830
226:AT2G01830
227:AT2G01830
228:AT2G01830
229:AT2G01830
230:AT2G01830
231:AT2G01830
232:AT2G01830
233:AT2G01830
234:AT2G01830
235:AT2G01830
236:AT2G01830
237:AT2G01830
238:AT2G01830
239:AT2G01830
240:AT2G01830
241:AT2G01830
242:AT2G01830
243:AT2G01830
244:AT2G01830
245:AT2G01830
246:AT2G01830
247:AT2G01830
248:AT2G01830
249:AT2G01830
250:AT2G01940
251:AT2G01940
252:AT2G02220
253:AT2G02950
254:AT2G03680
255:AT2G03680
256:AT2G03680
257:AT2G11810
258:AT2G11810
259:AT2G15290
260:AT2G15290
261:AT2G17420
262:AT2G18470
263:AT2G19560
264:AT2G20000
265:AT2G20000
266:AT2G20000
267:AT2G20000
268:AT2G20190
269:AT2G21170
270:AT2G22125
271:AT2G24270
272:AT2G24490
273:AT2G24790
274:AT2G25180
275:AT2G25180


In [14]:
grep -E -ino '\AT3G[0-9]{5}' Root-associated-Loci.txt > Chromosome3.txt
wc -l Chromosome3.txt
cat Chromosome3.txt

127 Chromosome3.txt
360:AT3G01650
361:AT3G01650
362:AT3G02680
363:AT3G05040
364:AT3G05120
365:AT3G05120
366:AT3G05120
367:AT3G05120
368:AT3G05630
369:AT3G05630
370:AT3G05630
371:AT3G06030
372:AT3G06810
373:AT3G06810
374:AT3G06810
375:AT3G06810
376:AT3G06810
377:AT3G07780
378:AT3G08040
379:AT3G08040
380:AT3G08710
381:AT3G09780
382:AT3G09840
383:AT3G09920
384:AT3G09920
385:AT3G10870
386:AT3G10870
387:AT3G11430
388:AT3G11430
389:AT3G13300
390:AT3G13300
391:AT3G13360
392:AT3G13360
393:AT3G13870
394:AT3G14000
395:AT3G14370
396:AT3G14370
397:AT3G14370
398:AT3G16150
399:AT3G16150
400:AT3G16150
401:AT3G16150
402:AT3G16785
403:AT3G16857
404:AT3G16857
405:AT3G16857
406:AT3G16857
407:AT3G17360
408:AT3G18780
409:AT3G18780
410:AT3G18780
411:AT3G19050
412:AT3G20630
413:AT3G20780
414:AT3G20840
415:AT3G20840
416:AT3G20840
417:AT3G22200
418:AT3G22200
419:AT3G22200
420:AT3G22400
421:AT3G22650
422:AT3G23050
423:AT3G23490
424:AT3G23490
425:AT3G23630
426:AT3G23630
427:AT3G23630
428:AT3G23630
429:AT3G23630


In [15]:
grep -E -ino '\AT4G[0-9]{5}' Root-associated-Loci.txt > Chromosome4.txt
wc -l Chromosome4.txt
cat Chromosome4.txt

73 Chromosome4.txt
330:AT4G17090
487:AT4G00710
488:AT4G01100
489:AT4G01370
490:AT4G02280
491:AT4G02280
492:AT4G02280
493:AT4G02780
494:AT4G05530
495:AT4G05530
496:AT4G05530
497:AT4G05530
498:AT4G05530
499:AT4G05530
500:AT4G07400
501:AT4G10310
502:AT4G10350
503:AT4G10350
504:AT4G10380
505:AT4G14550
506:AT4G14850
507:AT4G16760
508:AT4G17615
509:AT4G18750
510:AT4G18780
511:AT4G24400
512:AT4G24670
513:AT4G24670
514:AT4G24670
515:AT4G24670
516:AT4G24730
517:AT4G26080
518:AT4G26080
519:AT4G26455
520:AT4G26455
521:AT4G26610
522:AT4G26610
523:AT4G26690
524:AT4G26690
525:AT4G26690
526:AT4G26690
527:AT4G27630
528:AT4G29040
529:AT4G29040
530:AT4G29040
531:AT4G29130
532:AT4G29130
533:AT4G29130
534:AT4G29170
535:AT4G29210
536:AT4G29860
537:AT4G31920
538:AT4G32040
539:AT4G32700
540:AT4G32700
541:AT4G32700
542:AT4G32700
543:AT4G32700
544:AT4G33430
545:AT4G33430
546:AT4G33470
547:AT4G33480
548:AT4G33495
549:AT4G33650
550:AT4G34350
551:AT4G35090
552:AT4G35460
553:AT4G35920
554:AT4G36380
555:AT4G38130
5

In [16]:
grep -E -ino '\AT5G[0-9]{5}' Root-associated-Loci.txt > Chromosome5.txt
wc -l Chromosome5.txt
cat Chromosome5.txt

231 Chromosome5.txt
331:AT5G16150
559:AT5G01180
560:AT5G01180
561:AT5G01180
562:AT5G01410
563:AT5G01410
564:AT5G01410
565:AT5G01410
566:AT5G01410
567:AT5G01410
568:AT5G03150
569:AT5G03150
570:AT5G03280
571:AT5G03280
572:AT5G03280
573:AT5G03280
574:AT5G03540
575:AT5G03540
576:AT5G03730
577:AT5G03730
578:AT5G03730
579:AT5G03730
580:AT5G03730
581:AT5G03730
582:AT5G03860
583:AT5G03860
584:AT5G08020
585:AT5G08100
586:AT5G08100
587:AT5G08100
588:AT5G08100
589:AT5G08550
590:AT5G08550
591:AT5G09680
592:AT5G09810
593:AT5G09810
594:AT5G09810
595:AT5G10250
596:AT5G10330
597:AT5G10510
598:AT5G10510
599:AT5G11260
600:AT5G11260
601:AT5G11260
602:AT5G11260
603:AT5G11260
604:AT5G11260
605:AT5G11260
606:AT5G13290
607:AT5G13570
608:AT5G13570
609:AT5G13570
610:AT5G13570
611:AT5G13570
612:AT5G14420
613:AT5G14420
614:AT5G16530
615:AT5G16780
616:AT5G18170
617:AT5G18560
618:AT5G19040
619:AT5G19040
620:AT5G19040
621:AT5G19040
622:AT5G19040
623:AT5G19040
624:AT5G19660
625:AT5G20410
626:AT5G20490
627:AT5G20490


In [17]:
wc -l Root-associated-Loci.txt
echo "Chromosome1 has 206 lines, Chromosome2 has 151 lines, Chromosome3 has 127 lines, Chromosome4 has 73 lines and Chromosome5 has 231 lines"
echo "206+151+127+73+231=788"

788 Root-associated-Loci.txt
Chromosome1 has 206 lines, Chromosome2 has 151 lines, Chromosome3 has 127 lines, Chromosome4 has 73 lines and Chromosome5 has 231 lines
206+151+127+73+231=788


As we can see, all the outputs displayed a bunch of locus codes for each chromosome. In fact, if we add together the lines of all documents, the result is equal to the number of lines included in the original document. Therefore, the hypothesis is true in this dataset as we can find genes associated with roots on all chromosomes.    

Problem 10
The control experiment in problem 9 shows genes in every chromosome. 